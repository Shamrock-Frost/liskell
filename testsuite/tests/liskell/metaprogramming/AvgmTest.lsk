(defmodule AvgmTest _ (Liskell Avgm))

(defenv (lambda ((LskEnv e p t d)) (return (LskEnv (avgm-dspr trf_expr e) p t d))))

(define main
    (print (avgm 1 3 4 5 8)))
