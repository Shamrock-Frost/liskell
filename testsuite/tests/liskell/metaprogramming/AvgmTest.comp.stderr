
AvgmTest.lsk:3:71:
    Couldn't match expected type `TransformationMonad (HsExpr.LHsExpr RdrName.RdrName)'
	   against inferred type `(ParseTree -> a) -> (ParseTree -> a) -> a'
    In the second argument of `avgm-dspr', namely `e'
    In the first argument of `LskEnv', namely `avgm-dspr trf_expr e'
    In the first argument of `return', namely
	`LskEnv (avgm-dspr trf_expr e) p t d'
Loading package base ... linking ... done.
Loading package template-haskell ... linking ... done.
Loading package readline-1.0 ... linking ... done.
Loading package unix-1.0 ... linking ... done.
Loading package Cabal-1.1.7 ... linking ... done.
Loading package regex-base-0.71 ... linking ... done.
Loading package regex-posix-0.71 ... linking ... done.
Loading package regex-compat-0.71 ... linking ... done.
Loading package haskell98 ... linking ... done.
Loading package ghc-6.7 ... linking ... done.
